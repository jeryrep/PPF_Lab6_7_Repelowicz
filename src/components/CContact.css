.formContainer {
    margin: 1vw;
    text-align: left;
}

.formText {
    text-align: center;
    font-size: 3vw;
}

.inputText {
    width: 200px;
    display: block;
}