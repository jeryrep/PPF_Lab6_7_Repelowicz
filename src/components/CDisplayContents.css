.contentsFlex {
    background-color: #3d4151;
    display: flex;
    margin: 1vmin;
}
.contentsFlexImg {
    height: 20vmin;
    width: 25vmin;
}
.contentsFlexTxt {
    text-align: center;
}