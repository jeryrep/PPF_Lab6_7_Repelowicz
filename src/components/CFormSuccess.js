import React, {Component} from "react";
import './CFormSuccess.css';

export class CFormSuccess extends Component {
    constructor(props) {
        console.log("Konstruktor - wczytano dane")
        super(props);
        this.state = {
            name: '',
            surname: '',
            email: '',
            msg: 'Twoje dane są przesyłane...'
        };
    }

    static getDerivedStateFromProps(props) {
        console.log("Zapisano dane do states");
        return {
            name: props.name,
            surname: props.surname,
            email: props.email
        };
    }

    componentDidMount() {
        console.log("Po renderze");
        setTimeout(() => {
            this.setState({msg: "Postaramy się odpowiedzieć jak najszybciej na podany przez ciebie adres e-mail: " + this.state.email});
        }, 3000);
    }

    shouldComponentUpdate() {
        console.log("Komponent będzie aktualizowany")
        return true;
    }

    getSnapshotBeforeUpdate(prevProps, prevState) {
        console.log("Sprawdzam poprzedni zrzut");
        document.getElementById("thankYouText").title += "Przed aktualizacją wiadomość wyglądała tak: " + prevState.msg;
        return null;
    }

    componentDidUpdate(prevProps, prevState, snapshot) {
        console.log("Komponent został zaaktualizowany");
    }

    componentWillMount() {
        console.log("Komponent został usunięty");
    }

    validateSentData() {
        document.getElementById("result").innerHTML = this.state.name === this.props.name && this.state.surname === this.props.surname && this.state.email === this.props.email ? "Dane zostały poprawnie wysłane. Możesz opuścić stronę." : "Coś poszło nie tak, spróbuj ponownie później.";
    }

    render() {
        console.log("Renderuję wyjściowy komponent");
        return (
            <article className="formSuccess">
                <h1 className="testResult">
                    Dziękujemy za dane {this.props.name} {this.props.surname}!
                </h1>
                <p id="thankYouText">
                    {this.state.msg}
                </p>
                <p>
                    Kliknij poniższy przycisk by sprawdzić poprawność danych.
                </p>
                <button data-testid="Button" className="btn btn-dark" onClick={() => this.validateSentData(this.props, this.state)}>Sprawdź poprawność</button>
                <p data-testid="Result" id="result"></p>
            </article>
        );
    }
}