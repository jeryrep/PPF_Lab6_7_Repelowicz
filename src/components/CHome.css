.video {
    flex: 1 1 auto;
    height: 28vw;
    width: 48vw;
}