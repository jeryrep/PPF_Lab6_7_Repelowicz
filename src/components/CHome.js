import React from "react";
import {CVideoArticle} from "./CVideoArticle";

export const CHome = () => {
    return (
        <CVideoArticle/>
    );
}