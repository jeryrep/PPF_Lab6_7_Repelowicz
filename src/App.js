import './App.css';
import CHeader from "./components/CHeader";
import {CNavbar} from "./components/CNavbar";
import {CFooter} from "./components/CFooter";
import React, {Component} from "react";
import {CRoutes} from "./components/CRoutes";
import {BrowserRouter} from "react-router-dom";

class App extends Component {
    render() {
        return (
            <BrowserRouter>
                <div className="App">
                    <CHeader/>
                    <CNavbar/>
                    <CRoutes/>
                    {this.props.children}
                    <CFooter/>
                </div>
            </BrowserRouter>
        );
    }
}

export default App;
