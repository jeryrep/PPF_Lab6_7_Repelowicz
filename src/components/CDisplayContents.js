import {Component} from "react";
import './CDisplayContents.css';

export default class CDisplayContents extends Component {
    initializeJson(data) {
        this.data = () => JSON.parse(JSON.stringify(data));
        const imgContext = require.context('../res/', false, /\.jpg$/);
        let img = {};
        this.imgs = imgContext.keys().reduce((icons, file) => {
            const cname = imgContext(file).default;
            const label = file.slice(2, -4);
            img[label] = cname;
            return img;
        }, {});
    }

    resizeImage(param, e) {
        if (param === "enlarge") {
            e.target.style.height = "30vmin";
            e.target.style.width = "36vmin";
        } else {
            e.target.style.height = "20vmin";
            e.target.style.width = "23vmin";
        }
    }

    render() {
        return false;
    }
}