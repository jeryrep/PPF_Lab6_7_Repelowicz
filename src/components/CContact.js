import React from "react";
import './CContact.css';
import {CFormSuccess} from "./CFormSuccess";

export class CContact extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            name: '',
            surname: '',
            email: '',
            submitted: false
        };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({[event.target.name]: event.target.value});
    }

    handleSubmit(event) {
        event.preventDefault();
        this.setState({submitted: true});
    }

    renderFormInfo() {
        return <CFormSuccess name={this.state.name} surname={this.state.surname} email={this.state.email}/>;
    }

    render() {
        return (
            <article className="formContainer">
                <p className="formText display-4">
                    W celu skontaktowania się z naszą firmą wypęlnij poniższe dane kontaktowe
                </p>
                <form onSubmit={this.handleSubmit}>
                    <div className="form-group row">
                        <label htmlFor="name" className="col-sm-2 col-form-label">Imię:</label>
                        <div className="col-sm-10">
                            <input type="text" className="form-control inputText" placeholder="Jan" required name="name" onChange={this.handleChange}/>
                        </div>
                    </div>
                    <div className="form-group row">
                        <label htmlFor="surname" className="col-sm-2 col-form-label">Nazwisko:</label>
                        <div className="col-sm-10">
                            <input type="text" className="form-control inputText" placeholder="Kowalski" required name="surname" value={this.state.surname} onChange={this.handleChange}/>
                        </div>
                    </div>
                    <div className="form-group row">
                        <label htmlFor="email" className="col-sm-2 col-form-label">E-mail:</label>
                        <div className="col-sm-10">
                            <input type="email" className="form-control inputText" placeholder="jan.kowalski@gmail.com" required name="email" value={this.state.email} onChange={this.handleChange}/>
                        </div>
                    </div>
                    <div className="form-group row">
                        <div className="col-sm-10">
                            <input type="submit" className="btn btn-dark" required id="submit"/>
                        </div>
                    </div>
                </form>
                {this.state.submitted && this.renderFormInfo()}
            </article>
        );
    }
}