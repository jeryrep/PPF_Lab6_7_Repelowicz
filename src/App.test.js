import {fireEvent, render, screen} from '@testing-library/react';
import {CHome} from "./components/CHome";
import {CFormSuccess} from "./components/CFormSuccess";
import App from "./App";
import userEvent from "@testing-library/user-event";
import ShallowRenderer from 'react-test-renderer/shallow';
import {CContact} from "./components/CContact";
import Enzyme from 'enzyme';
import {mount, shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {CRoutes} from "./components/CRoutes";
import CDisplayContents from "./components/CDisplayContents";
import {CMaterials} from "./components/CMaterials";

Enzyme.configure({ adapter: new Adapter() });

test('CHome zawiera tekst', () => {
  render(<CHome />);
  const linkElement = screen.getByText("Od wielu lat zajmujemy się wszelkiego rodzaju metalami. Chcieliśmy zaprezentować film, który przybliży Ci, o czym jest nasza strona.");
  expect(linkElement).toBeInTheDocument();
});
test('następuje zmiana tekstu po wysłaniu formularza z opóźnieniem', () => {
  jest.useFakeTimers();
  render(<CFormSuccess/>);
  expect(setTimeout).toHaveBeenCalledTimes(1);
});
test('następuje zmiana tekstu po kliknięciu przycisku', () => {
  const {getByTestId} = render(<CFormSuccess name="Jan" surname="Kowalski" email="jan.kowalski@gmail.com"/>);
  const button = getByTestId("Button");
  const result = getByTestId("Result");
  expect(result).toHaveTextContent('');
  fireEvent.click(button);
  expect(result).toHaveTextContent("Dane zostały poprawnie wysłane. Możesz opuścić stronę.");
});
test('następuje przekierowanie po kliknięciu linku', () => {
  render(<App/>);
  const leftClick = {button: 0};
  userEvent.click(screen.getByText("Kontakt"), leftClick);
  expect(screen.getByText("W celu skontaktowania się z naszą firmą wypełnij poniższe dane kontaktowe")).toBeInTheDocument();
});
test('snapshot test', () => {
  const renderer = new ShallowRenderer();
   renderer.render(<App />);
   const result = renderer.getRenderOutput();
   expect(result).toMatchSnapshot();
});
test('logo poprawnie przekierowuje na stronę główną', () => {
  render(<App/>);
  const leftClick = {button: 0};
  userEvent.click(screen.getByText("Kontakt"), leftClick);
  expect(screen.getByText("W celu skontaktowania się z naszą firmą wypełnij poniższe dane kontaktowe")).toBeInTheDocument();
  userEvent.click(screen.queryByTestId("homeRouteLogo"), leftClick);
  expect(screen.getByText("Od wielu lat zajmujemy się wszelkiego rodzaju metalami. Chcieliśmy zaprezentować film, który przybliży Ci, o czym jest nasza strona.")).toBeInTheDocument();
});
test('dane są ładowane z formularza do states poprawnie', () => {
  const component = mount(<CContact/>);
  const input1 = component.find('input').at(0);
  const input2 = component.find('input').at(1);
  const input3 = component.find('input').at(2);
  input1.instance().value = 'Jan';
  input1.simulate('change');
  input2.instance().value = 'Kowalski';
  input2.simulate('change');
  input3.instance().value = 'jan.kowalski@gmail.com';
  input3.simulate('change');
  expect(component.state().name).toEqual('Jan');
  expect(component.state().surname).toEqual('Kowalski');
  expect(component.state().email).toEqual('jan.kowalski@gmail.com');
});
test('formularz się wysyła i wyświetla inny komponent z prawidłowymi danymi', async () => {
  const component = mount(<CContact/>);
  component.find('input').at(0).instance().value = 'Paweł';
  component.find('input').at(1).instance().value = 'Nowak';
  component.find('input').at(2).instance().value = 'pawel.nowak@gmail.com';
  component.find('form').simulate('submit');
  expect(component.find('h1').first().text()).toBe("Dziękujemy za dane " + component.state().name + " " + component.state().surname + "!");
});
test('zawiera dane do nawigacji pomiędzy podstronami', () => {
  const wrapper = shallow(<App/>);
  expect(wrapper.contains(<CRoutes/>)).toBe(true);
});
jest.mock('./components/CDisplayContents')
test('komponent wywołał konstruktor od komponentu, z którego dziedziczy', () => {
  const comp = new CMaterials(undefined);
  expect(CDisplayContents).toHaveBeenCalledTimes(1);
});