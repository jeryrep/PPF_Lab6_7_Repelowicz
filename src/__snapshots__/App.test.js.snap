// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <CHeader />
    <CNavbar />
    <CRoutes />
    <CFooter />
  </div>
</BrowserRouter>
`;
